### PR TITLE
Vkglscale

### PR DESCRIPTION
**Pull request by kd-11**
See https://github.com/Megamouse/rpcs3/commit/221c6496226b8718132040679413bf83fc4ffd8d
Branch needs rebase

### CODE REVIEW

**Comment by Megamouse** [comment]
Worst PR ever xD

**Comment by Megamouse** [comment]
Looks like you need to rebase now xD

**Comment by kd-11** [comment]
Done

File: rpcs3/Emu/RSX/rsx_utils.cpp [hunk 1]
```diff
@@ -135,60 +167,60 @@ namespace rsx
...
-		case video_aspect::_16_9: aspect_ratio = 16.0 / 9.0; break;
+		case CELL_VIDEO_OUT_ASPECT_16_9: return 16. / 9.;
+		case CELL_VIDEO_OUT_ASPECT_4_3: return 4. / 3.;
+		default: fmt::throw_exception("Invalid aspect ratio %d", aspect);
 		}
-		return aspect_ratio;
 	}
 
-	void avconf::upscale_to_aspect_ratio(int& width, int& height) const
+	size2u avconf::aspect_convert_dimensions(const size2u& image_dimensions) const
```
Comment (Megamouse) [inline rpcs3/Emu/RSX/rsx_utils.cpp:178]: why not just use a copy by value here and remove the result var?

Comment (Megamouse) [reply rpcs3/Emu/RSX/rsx_utils.cpp:178]: also, I feel like the old function name conveys what it actually does, while the new one is kinda cryptic

File: rpcs3/Emu/RSX/rsx_utils.cpp [hunk 2]
```diff
@@ -135,60 +167,60 @@ namespace rsx
...
-		}
-
-		double aspect_ratio;
-		switch (v_aspect)
-		{
-		case video_aspect::_4_3: aspect_ratio = 4.0 / 3.0; break;
-		case video_aspect::_16_9: aspect_ratio = 16.0 / 9.0; break;
+		case CELL_VIDEO_OUT_ASPECT_16_9: return 16. / 9.;
+		case CELL_VIDEO_OUT_ASPECT_4_3: return 4. / 3.;
+		default: fmt::throw_exception("Invalid aspect ratio %d", aspect);
```
Comment (Megamouse) [inline rpcs3/Emu/RSX/rsx_utils.cpp:174]: this change is pretty much fucking everything up.
The entire reason for this PR was the removed code above.
CELL_VIDEO_OUT_ASPECT_AUTO is a valid input here

File: rpcs3/Emu/RSX/rsx_utils.cpp (hunk 1 above)
Comment (kd-11) [reply rpcs3/Emu/RSX/rsx_utils.cpp:178]: How is scaling a 720p image to a 1080p or 4k size downscaling?
Accurate names would be something like aspect_ratio_correction_unconstrained and aspect_ratio_correction_constrained but that is too wordy. I'll think of something easier to read though.

File: rpcs3/Emu/RSX/rsx_utils.cpp (hunk 2 above)
Comment (kd-11) [reply rpcs3/Emu/RSX/rsx_utils.cpp:174]: It's not. Auto is a throw-away parameter when the game is making requests and it is never returned. Confirmed with hwtests. It is there for making the API nicer to devs so that they don't need to check system details every time.

File: rpcs3/Emu/RSX/rsx_utils.cpp (hunk 1 above)
Comment (Megamouse) [reply rpcs3/Emu/RSX/rsx_utils.cpp:178]: if you look closely you'll see that the function name was "upscale_to_aspect_ratio" xD

Comment (Megamouse) [reply rpcs3/Emu/RSX/rsx_utils.cpp:178]: I'd probably call it "expanded_to_aspect_ratio" or whatever

Comment (Megamouse) [reply rpcs3/Emu/RSX/rsx_utils.cpp:178]: like... Qt calls their size methods expandTo as well

File: rpcs3/Emu/RSX/rsx_utils.cpp (hunk 2 above)
Comment (Megamouse) [reply rpcs3/Emu/RSX/rsx_utils.cpp:174]: The member "aspect" is CELL_VIDEO_OUT_ASPECT_AUTO by default until set by cellVideoOutConfigure.
The logic above handles this case by using the cfg param as also seen in cellVideoOutConfigure.
I can easily reproduce crashes with this code removed.

Comment (kd-11) [reply rpcs3/Emu/RSX/rsx_utils.cpp:174]: If you have av resolution then aspect should be set to a real aspect. Auto should not be used that way, its the equivalent of passing a null parameter to mean "i don't care about the aspect just set the other stuff".
I'll add proper code to handle that.
